testsuite/classlibrary/TestCoreUGens.sc

```diff
@@ -645,6 +646,155 @@ TestCoreUGens : UnitTest {
 
 	}
 
+	test_line_startAndEndSampleAreOuputOverDur {
+		var durFrames = 5; // duration for the line to complete its trajectory, in frames
+		var renderFrames = durFrames + 1; // render one extra frame to check that the end value is reached
+		var startVal = 2;
+		var endVal = 20;
+		var tolerance = -100.dbamp;
+		var condvar = CondVar();
+
+		server.bootSync;
+
+		[\ar, \kr].do{ |rate|
```

Comment:
This might be easier to write as `[[\ar, \kr], [Line, XLine]].flop.do` to remove a level of indent. Just an idea, feel free to ignore.

Reply:
> [[\ar, \kr], [Line, XLine]].flop.do

This would actually interleave the vars so both rates aren't accessible from each UGen. And I think this separates the logic better (for each rate, iterate through each ugen...), and theoretically adding UGens to the list would be trivial, but wouldn't work in the flopping scheme.